src/odemis/util/transform.py

```diff
@@ -1,342 +1,815 @@
     """
+    Returns the 2x2 rotation matrix for a given angle.
 
-    def __init__(self, rotation=0, scaling=1, shear=0, translation=None):
-        self.rotation = rotation
-        self.scaling = scaling
-        self.shear = shear
-        self.translation = numpy.zeros(2) if translation is None else translation
+    Parameters
+    ----------
+    angle : float
+        Rotation angle in radians.
 
-    def __repr__(self):
-        return "<%s %r>" % (self.__class__.__name__, self.__dict__)
+    Returns
+    -------
+    matrix : 2x2 array
+        Rotation matrix.
+
+    Examples
+    --------
+    >>> _rotation_matrix_from_angle(numpy.pi / 2.)
+    array([[ 0., -1.],
+           [ 1.,  0.]])
+
+    """
+    ct = numpy.cos(angle)
+    st = numpy.sin(angle)
+    return numpy.array([(ct, -st),
+                        (st, ct)])
+
+
+def _rotation_matrix_to_angle(matrix):
+    """
+    Returns the angle for a given 2x2 rotation matrix.
+
+    Parameters
+    ----------
+    matrix : 2x2 array
+        Rotation matrix.
+
+    Returns
+    ------
+    angle : float
+        Rotation angle in radians.
+
+    Raises
+    ------
+    LinAlgError
+        If the supplied matrix is not a rotation matrix.
+    NotImplementedError
+        If the supplied matrix is not of size 2x2.
+
+    Examples
+    --------
+    >>> matrix = numpy.array([(0., -1.), (1., 0.)])
+    >>> _rotation_matrix_to_angle(matrix)
+    1.5707963267948966
 
-    @staticmethod
-    def _rotation_matrix_from_angle(angle):
+    """
+    _assertRotationMatrix(matrix)
+    if matrix.shape != (2, 2):
+        raise NotImplementedError('Can only handle 2x2 matrices')
+    return numpy.arctan2(matrix[1, 0], matrix[0, 0])
+
+
+def _optimal_rotation(x, y):
+    """
+    Returns the optimal rotation matrix between two zero mean point sets,
+    such that |Rx-y|^2 is minimized.
+
+    Parameters
+    ----------
+    x : list of tuples
+        List of coordinates with zero mean in the source frame of reference.
+    y : list of tuples
+        List of coordinates with zero mean in the destination frame of
+        reference.
+
+    Returns
+    -------
+    matrix : 2x2 array
+        Rotation matrix.
+
+    Raises
+    ------
+    AssertionError
+        If either the source or destination point set is not zero mean.
+
+    Examples
+    --------
+    >>> x = [(1., 0.), (-1., 0.)]
+    >>> y = [(0., 2.), (0., -2.)]
+    >>> _optimal_rotation(x, y)
+    array([[ 0., -1.],
+           [ 1.,  0.]])
+
+    """
+    assert numpy.allclose(numpy.average(x, axis=0), numpy.zeros(2))
+    assert numpy.allclose(numpy.average(y, axis=0), numpy.zeros(2))
+    H = numpy.dot(numpy.transpose(x), y)
+    U, _, V = numpy.linalg.svd(H, full_matrices=False)  # H = USV (not V')
+    if numpy.linalg.det(U) * numpy.linalg.det(V) < 0.0:
+        U[:, -1] = -U[:, -1]
+    return numpy.dot(V.T, U.T)
+
+
+class GeometricTransform(object):
+    """Base class for geometric transformations."""
```

Comment:
Updated the code and I used:
```
@property
@abstractmethod
def transformation_matrix(self):
    pass
```
because `@abstractproperty` [is deprecated](https://docs.python.org/3/library/abc.html#abc.abstractproperty).